src/Controller/IdentityController.php

```diff
@@ -182,6 +183,24 @@ public function ssoReturnAction()
         return $response;
     }
 
+    /**
+     * Generates a stepup requets id.
+     *
+     * The request ID should be generated by
+     * hashing the application and machine name and timestamp to a hexadecimal string.
+     *
+     * @param Request $request
+     *
+     * @return array|string
+     */
+    private function getOrGenerateStepupRequestId(Request $request)
+    {
+        if ($request->headers->has('X-Stepup-Request-Id')) {
+            return $request->headers->get('X-Stepup-Request-Id');
+        }
+        return bin2hex(implode([$request->getHost(), gethostname(), time()]));
```

Comment:
Right?

Reply:
My suggestion is to validate this with one of the product owners.

`\Surfnet\StepupBundle\Request\RandomRequestIdGenerator::generateRequestId` uses `md5(openssl_random_pseudo_bytes(50));` to generate a RequestID.